academic_observatory_workflows/s5cmd.py

```diff
@@ -0,0 +1,172 @@
+import re
+import shlex
+import logging
+from typing import List, Tuple, Union
+from subprocess import Popen, PIPE
+from contextlib import contextmanager
+from tempfile import NamedTemporaryFile
+
+
+class S5CmdCpConfig:
+    """Configuration for S5Cmd cp command
+
+    :param flatten_dir: Whether to flatten the directory structure
+    :param no_overwrite: Whether to not overwrite files if they already exist
+    :param overwrite_if_size: Whether to overwrite files only if source size differs
+    :param overwrite_if_newer: Whether to overwrite files only if source is newer
+    """
+
+    def __init__(
+        self,
+        flatten_dir: bool = False,
+        no_overwrite: bool = False,
+        overwrite_if_size: bool = False,
+        overwrite_if_newer: bool = False,
+    ):
+        self.flatten_dir = flatten_dir
+        self.no_overwrite = no_overwrite
+        self.overwrite_if_size = overwrite_if_size
+        self.overwrite_if_newer = overwrite_if_newer
+
+    def __str__(self):
+        cfg = [
+            "--flatten" if self.flatten_dir else "",
+            "--no-clobber" if self.no_overwrite else "",
+            "--if-size-differ" if self.overwrite_if_size else "",
+            "--if-source-newer" if self.overwrite_if_newer else "",
+        ]
+        cfg = [i for i in cfg if i]  # Remove empty strings
+        return " ".join(cfg)
+
+
+class S5Cmd:
+    def __init__(
+        self,
+        access_credentials: Tuple[str, str],
+        out_stream: str = PIPE,
```

Comment:
The `out_stream` variable should have a type of `Union[None, int, TextIO]` which matches the `Popen` `stdout` and `stderr` types:

```python
out_stream: Union[None, int, TextIO] = PIPE
```

You can import `TextIO` from `from typing import TextIO`.